npm/webpack-dev-server/src/loader.ts

```diff
@@ -18,7 +18,7 @@ const makeImport = (file: Cypress.Cypress['spec'], filename: string, chunkName:
   const magicComments = chunkName ? `/* webpackChunkName: "${chunkName}" */` : ''
 
   return `"${filename}": {
-    shouldLoad: () => decodeURI(document.location.pathname).includes("${file.absolute}"),
+    shouldLoad: () => (new URL(document.location)).searchParams.get("specPath") === "${file.absolute}",
```

Comment:
Maybe a teensy-tiny perf improvement: could pass `document.location.search` into `URLSearchParams` rather than parsing the full URL to pull out the params

```ts
new URLSearchParams(document.location.search).get("specPath")
```

Reply:
I'll give this a go, see how CI goes! Should be fine